packages/ember-data/lib/system/store.js

```diff
@@ -1445,8 +1447,9 @@ Store = Service.extend({
   },
 
   modelFactoryFor: function(key) {
-    if (this.container.has('model:' + key)) {
-      return this.container.lookupFactory('model:' + key);
+    var singularized = singularize(key);
```

Comment:
That is a good point! looking into it now. Thanks for taking the time to review! Makes me more confident that we'll catch more things before this gets merged in.